packages/next/src/build/index.ts

```diff
@@ -396,12 +397,7 @@ export default async function build(
       NextBuildContext.config = config
 
       let configOutDir = 'out'
-      if (config.output === 'export' && config.distDir !== '.next') {
-        // In the past, a user had to run "next build" to generate
-        // ".next" (or whatever the distDir) followed by "next export"
-        // to generate "out" (or whatever the outDir). However, when
-        // "output: export" is configured, "next build" does both steps.
-        // So the user-configured distDir is actually the outDir.
+      if (hasCustomExportOutput(config)) {
```

Comment:
this logic needs to be duplicated in webpack, so the manifest are pointing to the dist folder or temporary output folder (`.next`) instead of the actual output folder.

For output mode:
In `.next`, it could contain those manifests;
In `out` foler, it's pure text content like html, so manfest shouldn't point there